glean-core/src/upload/mod.rs

```diff
@@ -98,7 +106,12 @@ impl RateLimiter {
         }
 
         if self.count == self.max_count {
-            return RateLimiterState::Throttled;
+            let remaining = self.interval.as_millis() - self.elapsed().as_millis();
+            return RateLimiterState::Throttled(
+                remaining
+                    .try_into()
+                    .unwrap_or(self.interval.as_secs() * 1000),
```

Comment:
Why doing the conversion manually instead of using `as_millis`?

Reply:
`as_millis` returns a `u128` which we need to `try_into` a `u64`. If that is unsuccesfull I make it default to the interval value through `as_secs` which already returns a `u64`.

Reply:
Ah, someone would need to use a huuuuge interval :D 
But that also means the first try_into is unlikely to fail, so we can keep it as is.